tests/ui/typeck/issue-31173.stderr

```diff
@@ -31,6 +31,7 @@ LL | |             found_e = true;
 LL | |             false
 LL | |         })
 LL | |         .cloned()
+   | |          -------- method `collect` found for this std::iter::TakeWhile<&mut std::vec::IntoIter<u8>, [closure@$DIR/issue-31173.rs:7:21: 7:25]>
```

Comment:
Maybe we should to not show this label for this case, as it is somewhat redundant: the following label already imply that `collect` is available for `cloned`, except for the trait bounds. But then again it probably is ok either way.

Reply:
Let me know your decision. I do not understand the context of the typecheck yet!